### ISSUE TITLE
Use of deprecated ICU functions

### ISSUE THREAD

**Issue by lucasvr** [LinearTapeFileSystem/ltfs#45]
# Description

LTFS uses deprecated ICU functions for path normalization checks. For instance:
```
pathname.c: In function ‘_pathname_normalize_nfc_icu’:
pathname.c:661:2: warning: ‘unorm_quickCheck_60’ is deprecated [-Wdeprecated-declarations]
  if (unorm_quickCheck(src, -1, UNORM_NFC, &err) == UNORM_YES) {
  ^~
In file included from /usr/local/include/unicode/platform.h:25:0,
                 from /usr/local/include/unicode/ptypes.h:52,
                 from /usr/local/include/unicode/umachine.h:46,
                 from /usr/local/include/unicode/utypes.h:38,
                 from /usr/local/include/unicode/uchar.h:28,
                 from pathname.c:62:
/usr/local/include/unicode/unorm.h:240:1: note: declared here
 unorm_quickCheck(const UChar *source, int32_t sourcelength,
 ^
```

Using the alternative functions declared in `unorm2.h` seems to be the best approach moving forward.

# Environment

* OS:              Linux
* Distribution:    GoboLinux 016.01
* Architecture:    x86_64
* Additional info: LibICU4C 60.2

# How to recreate

Compile LTFS with a recent version of LibICU4C (such as 60.2).

**Comment by piste-jp-ibm** [comment]
@lucasvr , I believe this was fixed by unotm2 patch in #52.

Please close this if you confirm the patch is valid.

**Comment by rzerres** [comment]
@lucasvr 
@piste-jp-ibm 

I reworked the ICU patch to make ICU 6x selectable.
It compiles, but when mounting a tape i get a memory-error.
This needs to be addressed

**Comment by lucasvr** [comment]
@piste-jp-ibm, this issue can be closed once PR #60 is merged.

**Comment by lucasvr** [comment]
Thanks for reviewing the patch, @piste-jp-ibm. I am closing this issue now.
